packages/core/src/sync-realtime/service.ts

```diff
@@ -332,11 +332,21 @@ export const handleBlock = async (
 
   // "eth_getLogs" calls can be skipped if a negative match is given from "logsBloom".
   const shouldRequestLogs =
-    service.factoryLogSources.length > 0 ||
     newHeadBlock.logsBloom === zeroLogsBloom ||
     isMatchedLogInBloomFilter({
       bloom: newHeadBlock.logsBloom,
-      logFilters: service.logSources.map((s) => s.criteria),
+      logFilters: [
+        ...service.logSources.map((s) => s.criteria),
+        // child address logs
+        ...service.factoryLogSources.map((s) => ({
+          address: s.criteria.address,
+          topics: [s.criteria.eventSelector],
+        })),
```

Comment:
And this is the filter for logs announcing new child addresses?

It's weird to me that `s.criteria.address` is the factory contract address, while `s.criteria.topics` is the child contract topics. Not a huge deal though, makes more sense in other contexts.